spec/yaml/swiftnav/sbp/observation.yaml

```diff
@@ -80,32 +103,68 @@ definitions:
             type: CarrierPhase
             units: cycles
             desc: Carrier phase observation with typical sign convention.
+        - D:
+            type: Doppler
+            units: Hz
+            desc: Doppler observation with typical sign convention.
         - cn0:
             type: u8
-            units: dB Hz * 4
+            units: dB Hz * 0.25
             desc: Carrier-to-Noise density
         - lock:
-            type: u16
+            type: u8
             desc: |
-              Lock indicator. This value changes whenever a satellite
-              signal has lost and regained lock, indicating that the
-              carrier phase ambiguity may have changed.
+              Lock timer. This value gives an indication of the time
+              for which a signal has maintained continuous phase lock.
+              Whenever a signal has lost and regained lock, this 
+              value is reset to zero. It is encoded according to DF402 from
+              the RTCM 10403.2 Amendment 2 specification.  Valid values range 
+              from 0 to 15 and the most significant nibble is reserved for future use.
+              The minimum and maximum lock times for each value of the field can be described by the
+              following piecewise function.  Given the lock value, l, the minimum lock time 
+              is given by 2 ^ (l + 4) ms and the maximum lock time is given by 2 ^ (l + 5) ms 
+              provided n is not 0.  If n is 0 the lower range is given to be 0 ms.
+              Conversely, given a lock time (t) in milliseconds, the field value is given by 
+              floor(log_2(t) - 4) when t is greater than 32 ms or 0 if (t) is less than 32 ms.
+        - flags:
+            type: u8
+            desc: |
+              Measurement status flags. A bit field of flags providing the
+              status of this observation.  If this field is 0 it means only the Cn0
+              estimate for the signal is valid.
+            fields:
+              - 3-7:
+                  desc: Reserved
+              - 2:
+                  desc: half-cycle ambiguity
+                  values:
+                      - 0: Half cycle phase ambiguity unresolved
+                      - 1: Half cycle phase ambiguity resolved
+              - 1:
+                  desc: Carrier valid
+                  values:
+                      - 0: Invalid Carrier Phase measurement
+                      - 1: Valid Carrier Phase measurement (half cycle ambiguity may exist)
+              - 0:
+                  desc: Code valid
+                  values:
+                      - 0: Code phase invalid
+                      - 1: Valid code phase and coarse TOW decoded
         - sid:
-            type: GnssSignal
-            desc: GNSS signal identifier
+            type: GnssSignal16
```

Comment:
Yeah, I'd just bring this in here for now and call it `ObsGnssSignal` until we can move all the other messages on to it, if that's desired.